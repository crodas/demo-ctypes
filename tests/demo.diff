001+ array(13) {
001- hi there
002+   [0]=>
003+   bool(true)
004+   [1]=>
005+   object(CTypes\Resource)#1 (0) {
006+   }
007+   [2]=>
008+   int(14)
009+   [3]=>
010+   resource(4) of type (ctypes)
011+   [4]=>
012+   resource(5) of type (ctypes)
013+   [5]=>
014+   int(16785408)
015+   [6]=>
016+   int(16793600)
017+   [7]=>
018+   int(16908288)
019+   [8]=>
020+   int(17301504)
021+   [9]=>
022+   int(50339840)
023+   [10]=>
024+   string(16) "/lib64/libc.so.6"
025+   [11]=>
026+   object(CTypes\FunctionProxy)#4 (0) {
027+   }
028+   [12]=>
029+   bool(true)
030+ }
031+ destroying
032+ resource(6) of type (ctypes)
033+ destroying
034+ resource(7) of type (ctypes)
035+ object(CTypes\Library)#3 (0) {
036+ }
037+ object(CTypes\Library)#3 (0) {
038+ }
039+ Calling 33261d70 (isalpha) from /lib64/libc.so.6