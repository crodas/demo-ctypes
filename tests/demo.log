
---- EXPECTED OUTPUT
hi there
---- ACTUAL OUTPUT

---- FAILED
