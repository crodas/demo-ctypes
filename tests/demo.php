<?php
use CTypes\Library;
use CTypes\Resource;

$weird_type = new Resource("some name");
$weird_type->setDestructor(function($w) {
    echo "destroying\n";        
    var_dump($w);
});

class Foobar {}

$c = new Library("/lib64/libc.so.6");
var_dump(array(
    $c, 
    $weird_type, 
    $weird_type->getResourceId(),
    $weird_type->getResourceDemo(),
    $weird_type->getResourceDemo(),
    Library::tInteger,
    Library::tChar,
    Library::tString,
    Library::tFloat,
    Library::tPtr | Library::tInteger,
    $c->getLibraryPath(),
    $c->getFunction('isalpha', $weird_type, array(Library::tChar, $weird_type->getResourceId()| Library::tPtr  )),
    $c->getFunction('isalpha', Library::tBool, array(Library::tChar)),
));

