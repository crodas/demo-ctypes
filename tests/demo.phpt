--TEST--
Some test
--FILE--
<?php
use CTypes\Library;
use CTypes\Resource;

$weird_type = new Resource("some name");
$weird_type->setDestructor(function($w) {
    echo "destroying\n";        
    var_dump($w);
});

class Foobar {}

$c = new Library("/lib64/libc.so.6");

var_dump(array(
    $c, 
    $weird_type, 
    $weird_type->getResourceId(),
    $weird_type->getResourceDemo(),
    $weird_type->getResourceDemo(),
    Library::tInteger,
    Library::tChar,
    Library::tString,
    Library::tFloat,
    Library::tPtr | Library::tInteger,
    $c->getLibraryPath(),
    $var = $c->getFunction('isalpha', $weird_type, array(Library::tChar, $weird_type->getResourceId()| Library::tPtr  )),
    is_callable( $c->getFunction('isalpha', Library::tBool, array(Library::tChar)) ),
));
$memory = new Resource;
$malloc = $c->getFunction('malloc', $memory, array(Library::tInteger));
$memory->setDestructor($c->getFunction('free', NULL, array($memory)));

$bytes = $malloc(250);
unset($bytes);


var_dump($var->getLibrary());
var_dump($var->getLibrary());
$var();
--EXPECT--
hi there
