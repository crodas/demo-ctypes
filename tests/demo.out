2 arguments
isalpha
1 arguments
isalpha
array(13) {
  [0]=>
  object(CTypes\Library)#3 (0) {
  }
  [1]=>
  object(CTypes\Resource)#1 (0) {
  }
  [2]=>
  int(14)
  [3]=>
  resource(4) of type (ctypes)
  [4]=>
  resource(5) of type (ctypes)
  [5]=>
  int(16785408)
  [6]=>
  int(16793600)
  [7]=>
  int(16908288)
  [8]=>
  int(17301504)
  [9]=>
  int(50339840)
  [10]=>
  string(16) "/lib64/libc.so.6"
  [11]=>
  object(CTypes\FunctionProxy)#4 (0) {
  }
  [12]=>
  bool(true)
}
release 14
destroying
resource(6) of type (ctypes)
release 14
destroying
resource(7) of type (ctypes)
1 arguments
malloc
1 arguments
free
Calling 5fe0200 (malloc) from /lib64/libc.so.6
object(CTypes\Library)#3 (0) {
}
object(CTypes\Library)#3 (0) {
}
Calling 5f90430 (isalpha) from /lib64/libc.so.6